sdks/opik_optimizer/src/opik_optimizer/demo/datasets.py

```diff
@@ -28,33 +28,44 @@ def get_or_create_dataset(
         "rag_hallucinations",
     ],
     test_mode: bool = False,
+    seed: int = 42,
 ) -> opik.Dataset:
-    """Get or create a dataset from HuggingFace."""
+    """Get or create a dataset from HuggingFace, using the provided seed for sampling."""
     try:
```

Comment:
In general, try/except for large blocks like this are a bad idea. They hide bugs. Try to block just the small lines that can fail.